### PR TITLE
Fixed Makefile so that the dev-switch rule works with a shell different than bash

### PR DESCRIPTION
**Pull request by esope**
The command "[[ ... ]]" is supported by bash, but not by sh or dash
(these shell are used by default in Ubuntu, for example).

Signed-off-by: Benoît Montagu <benoit.montagu@inria.fr>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -67,7 +67,7 @@ dev-deps:
 dev-switch:
 	opam update
 # Ensuring that either a dev switch already exists or a new one is created
-	[[ $(shell opam switch show) == $(shell pwd) ]] || \
+	test "$(shell opam switch show) == $(shell pwd)" || \
```
Comment (nojb) [inline Makefile:70]: Seems strange to quote the whole comparison term. Shouldn't you be quoting each one of the operands?

Comment (nojb) [reply Makefile:70]: Also, if you worry about `sh` compatibility, I guess it is better to use `=` rather than `==` (they are synonyms under `bash`).

Comment (esope) [reply Makefile:70]: Agreed. I'll update asap.

**Review by nojb — APPROVED** [review]
LGTM (but perhaps one of our `Makefile` experts want to take a look cc @dra27 @shindere)

**Comment by shindere** [comment]
No time to review right now but many thanks for having mentionned me!

**Review by nojb — APPROVED** [review]

**Review by nojb — APPROVED** [review]
LGTM

**Pull request merged**